src/datastore/database.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <vector>
+#include "src/datastore/souvenir.hpp"
+#include "src/datastore/stadium.hpp"
+#include "src/datastore/team.hpp"
+#include "src/utils/map.hpp"
+
+/**
+ * @class Database class
+ *
+ * This class holds all of the programs data.
+ * It contains a map of teams and a map of stadiums.
+ * Each stadium has a vector of souvenirs within it's class.
+ *
+ * All the members aside from getEnumValue are static
+ */
+class Database
+{
+public:
+    /* Static methods to Load and Save from/to file */
+    static void loadFromFile(const std::string&);
+    static void saveToFile(const std::string&);
+
+    /* Getters */
+    const static nstd::map<int,Team> getTeams();
+    const static nstd::map<int,Stadium> getStadiums();
+    static Team& findTeamById(int);
+    static Stadium& findStadiumById(int);
+    static std::vector<Team> getTeamsVector();
+    static std::vector<Stadium> getStadiumsVector();
+
+private:
+    /* Private Constructor */
+    Database();
+
+    /* Private string to Enum method*/
+    template<typename Container, typename Enum>
+    Enum getEnumValue(const Container& ,const std::string& , Enum);
```

Comment:
Can we get a `const` qualifier on this function?

Reply:
To return a const enum?

Reply:
No I'm talking about making the function constant.
`Enum getEnumValue(...) const;`

Reply:
k